File: sample/src/main/java/com/readdle/swiftjava/sample/HashMapTest.kt
package com.readdle.swiftjava.sample

import com.readdle.codegen.anotation.SwiftFunc
import com.readdle.codegen.anotation.SwiftGetter
import com.readdle.codegen.anotation.SwiftValue

@SwiftValue
enum class MapKey(val rawValue: String) {
    one("one"),
    two("two");
}
//
//@SwiftValue
//data class Mapper(
//    var map: HashMap<MapKey, Int> = hashMapOf()
//)

@SwiftValue
class Mapper private constructor() {
    @get:SwiftGetter
    val map: HashMap<MapKey, Int> external get

    companion object {
        @JvmStatic
        @SwiftFunc("init(map:)")
        external fun init(
            map: HashMap<MapKey, Int>
        ): Mapper
    }
}

File: sample/src/main/swift/Sources/SampleAppCore/HashMapTest.swift
import Foundation

public enum MapKey: String, Codable {
    case one
    case two
}

public typealias MapType = [MapKey: Int]

public struct Mapper: Codable {
    public let map: MapType

    public init(map: MapType) {
        self.map = map
    }
}
